kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/challenges/foodproducers/Processor.java

```diff
@@ -0,0 +1,30 @@
+package com.kodilla.good.patterns.challenges.foodproducers;
+
+public class Processor {
+
+    private SaleInformation saleInformation;
+    private ProducerProcess producerProcess;
+    private StoreService storeService;
+
+    public Processor(SaleInformation saleInformation, ProducerProcess producerProcess, StoreService storeService) {
+
+        this.saleInformation = saleInformation;
+        this.producerProcess = producerProcess;
+        this.storeService = storeService;
+    }
+
+    public boolean process(Order order) {
+
+        storeService.showAllAvailableProduct();
+        boolean isSold = producerProcess.process(order.getUser(), order.getProduct(), storeService);
```

Comment:
you can have many producers, but here you use only one (producer which was provided in constructor) - it is serious bug ;)